src/scripts/main.js

```diff
@@ -1,3 +1,172 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', function () {
+  const employeeTable = document.querySelector('table');
+  let currentSort = { column: null, order: 'asc' };
+  let selectedRow = null;
+
+  const headers = document.querySelectorAll('th');
+
+  headers.forEach((header, index) => {
+    header.addEventListener('click', () => {
+      const order =
+        currentSort.column === index && currentSort.order === 'asc'
+          ? 'desc'
+          : 'asc';
+
+      sortTable(employeeTable, index, order);
+      currentSort = { column: index, order: order };
+    });
+  });
+
+  function sortTable(table, columnIndex, order) {
+    const rows = Array.from(table.tBodies[0].rows);
+    const isNumeric = columnIndex === 3 || columnIndex === 4;
```

Comment:
The `isNumeric` variable is hard-coded to only consider columns 3 and 4 as numeric. This could lead to issues if the table structure changes in the future. It would be better to find a more flexible way to determine if a column should be sorted numerically or not.